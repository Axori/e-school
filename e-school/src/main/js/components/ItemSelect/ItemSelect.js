import React from "react";
import Select from "../Select/Select";
import Loader from "../Loader/Loader";

const ItemSelect = ({id, options, onChange, selected}) => {
    return <div className="d-flex justify-content-center">
        {options != null ? <Select
                id={id}
                disabled={true}
                selected={selected}
                onChange={onChange}
                options={options}/>
            : <Loader/>}
    </div>
}

export default ItemSelect;