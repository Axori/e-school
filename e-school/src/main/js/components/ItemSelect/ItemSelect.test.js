import React from 'react';
import ItemSelect from "./ItemSelect";
import renderer from 'react-test-renderer';

describe("ItemSelect", () => {
    const options = [
        {
            label: "Option 1",
            value: "option1"
        },
        {
            label: "Option 2",
            value: "option2"
        },
        {
            label: "Option 3",
            value: "option3"
        }
    ];
    const defaultProps = {
        id: "ItemSelectId",
        options,
        onChange: jest.fn(),
        selected: options[0].value
    }

    const getComponent = (props = {}) => renderer.create(<ItemSelect {...defaultProps} {...props} />)

    it("should render select when options are defined", () => {
        const component = getComponent();

        expect(component.toJSON()).toMatchSnapshot();
    })

    it("should render loading when options are undefined", () => {
        const component = getComponent({options: undefined, selected: undefined});

        expect(component.toJSON()).toMatchSnapshot();
    })
})