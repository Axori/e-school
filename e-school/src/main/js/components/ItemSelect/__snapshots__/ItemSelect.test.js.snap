// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemSelect should render loading when options are undefined 1`] = `
<div
  className="d-flex justify-content-center"
>
  <div
    className="spinner-border"
    role="status"
  >
    <span
      className="visually-hidden"
    >
      Loading...
    </span>
  </div>
</div>
`;

exports[`ItemSelect should render select when options are defined 1`] = `
<div
  className="d-flex justify-content-center"
>
  <select
    aria-label="Default select example"
    className="form-select"
    id="ItemSelectId"
    onChange={[Function]}
  >
    <option
      value="option1"
    >
      Option 1
    </option>
    <option
      value="option2"
    >
      Option 2
    </option>
    <option
      value="option3"
    >
      Option 3
    </option>
  </select>
</div>
`;
